test/Dialect/RTL/canonicalization.mlir

```diff
@@ -766,6 +766,43 @@ rtl.module @shrs_fold2() -> (i12) {
   rtl.output %0 : i12
 }
 
+// CHECK-LABEL:  rtl.module @icmp_fold_1bit_eq1(%arg: i1) -> (i1) {
+// CHECK-NEXT:    %true = comb.constant true
+// CHECK-NEXT:    %0 = comb.xor %arg, %true : i1
+// CHECK-NEXT:    rtl.output %0 : i1
+      
+rtl.module @icmp_fold_1bit_eq1(%arg: i1) -> (i1) {
+  %c1 = comb.constant 0 : i1
+  %0 = comb.icmp eq  %c1, %arg : i1
+  rtl.output %0 : i1
+}
+ // CHECK-LABEL: rtl.module @icmp_fold_1bit_eq2(%arg: i1) -> (i1) {
+ // CHECK-NEXT:    rtl.output %arg : i1
+
+rtl.module @icmp_fold_1bit_eq2(%arg: i1) -> (i1) {
+  %c1 = comb.constant 1 : i1
+  %0 = comb.icmp eq   %c1, %arg : i1
+  rtl.output %0 : i1
+}
+// CHECK-LABEL:  rtl.module @icmp_fold_1bit_neq1(%arg: i1) -> (i1) {
+ // CHECK-NEXT:    rtl.output %arg : i1
+      
+rtl.module @icmp_fold_1bit_neq1(%arg: i1) -> (i1) {
+  %c1 = comb.constant 0 : i1
+  %0 = comb.icmp ne  %c1, %arg : i1
+  rtl.output %0 : i1
+}
+ // CHECK-LABEL: rtl.module @icmp_fold_1bit_neq2(%arg: i1) -> (i1) {
+// CHECK-NEXT:    %true = comb.constant true
+// CHECK-NEXT:    %0 = comb.xor %arg, %true : i1
+// CHECK-NEXT:    rtl.output %0 : i1
+
+rtl.module @icmp_fold_1bit_neq2(%arg: i1) -> (i1) {
+  %c1 = comb.constant 1 : i1
+  %0 = comb.icmp ne  %c1, %arg  : i1
+  rtl.output %0 : i1
+}
```

Comment:
Nit: I'd make the comments here look like the comments above:

1. No space between the comment and the test it's associated with
1. One space after each test
1. Keep everything flush left (remove any of the extra spaces included on the left)

Reply:
Other nit, fold all the bodies into one module.  No need to have a separate module per condition.  I realize this is somewhat ad-hoc, but for simple checks of the same set of folds, it seems nicer to not make lots of extra text.

Reply:
Updated the test cases into a single module